### PR TITLE
MGMT-8337: Remove unsused hostname from Agent CRD

### PR DESCRIPTION
**Pull request by rollandf**
# Assisted Pull Request

## Description
The current hostname is already available in Status.Inventory.HostName.

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @filanov 
/cc @avishayt 
/cc @eranco74 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/controllers/agent_controller.go [hunk 1]
```diff
@@ -703,6 +703,7 @@ func (r *AgentReconciler) updateInventory(log logrus.FieldLogger, host *models.H
 		log.WithError(err).Errorf("Failed to unmarshal host inventory")
 		return err
 	}
+	agent.Status.Hostname = inventory.Hostname
```
Comment (avishayt) [inline internal/controller/controllers/agent_controller.go:706]: What's the point of agent.Status.Hostname if it's the same as agent.Status.Inventory.Hostname?

Comment (rollandf) [reply internal/controller/controllers/agent_controller.go:706]: So what is https://issues.redhat.com/browse/MGMT-8337 about?

You want to remove the field?

Comment (rollandf) [reply internal/controller/controllers/agent_controller.go]: I remove the field from the CRD

**Review by avishayt — APPROVED** [review]

**Review by filanov — APPROVED** [review]

**Pull request merged**